Assets/Code/Components/AI/Behaviour/Nodes/SelectorNode.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace Assets.Code.Components.AI.Behaviour.Nodes
+{
+    class SelectorNode : BehaviourNode
+    {
+        private BehaviourNode[] nodes;
+        private int index;  // The current routine index
+
+        public SelectorNode(BtContext context, BehaviourNode[] routines) : base(context)
+        {
+            if (routines == null || routines.Length == 0)
+            {
+                throw new ArgumentException("cannot be null or empty", "routines");
+            }
+
+            this.nodes = routines;
+            index = 0;
+        }
+
+        public override void Start()
+        {
+            base.Start();
+            nodes[index].Start();
+        }
+
+        public override void Act()
+        {
+            base.Act();
+            nodes[index].Act();
+
+            if (nodes[index].HasSucceeded)
```

Comment:
El selector es como un "OR" y el sequence un "AND". Para que que un selector termine con "Success" al menos uno de sus nodos hijos tiene que dar success. Con el sequence, todos los hijos tienen que dar success. Yo creo que si da para dos clases para mantener separada la lógica.